internal/impl/postgresql/pglogicalstream/snapshotter.go

```diff
@@ -96,51 +93,37 @@ func (s *Snapshotter) setTransactionSnapshotName(snapshotName string) {
 	s.snapshotName = snapshotName
 }
 
-func (s *Snapshotter) prepare() error {
+func (s *Snapshotter) prepare(ctx context.Context) error {
 	if s.snapshotName == "" {
 		return errors.New("snapshot name is not set")
 	}
-
-	if _, err := s.pgConnection.Exec("BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ;"); err != nil {
-		return err
+	if s.readerTxn != nil {
+		return errors.New("reader txn already open")
 	}
-
+	// Use a background context because we explicitly want the Tx to be long lived, we explicitly close it in the close method
+	tx, err := s.pool.BeginTx(context.Background(), &sql.TxOptions{ReadOnly: true, Isolation: sql.LevelRepeatableRead})
```

Comment:
Should we use `ctx` here?

Reply:
I have a comment above but using the passed in context can cancel the transaction early, we want to explicitly control the lifetime of the transaction instead

Reply:
Ugh... Please ignore me 🙈 Sounds good!